modules/ibm-z-rhel-kvm-host-recommendations.adoc

```diff
@@ -0,0 +1,181 @@
+
+== Avoid virtual SCSI devices
+
+Configure virtual SCSI devices only if you need to address the device through SCSI-specific interfaces. Configure disk space as virtual block devices rather than virtual SCSI devices, regardless of the backing on the host.
+
+However, you might need SCSI-specific interfaces for:
+
+* A LUN for a SCSI-attached tape drive on the host.
+
+* A DVD ISO file on the host file system that is mounted on a virtual DVD drive.
+
+== Configure guest caching for disk
+
+Configure your disk devices to do caching by the guest and not by the host.
+
+Ensure that the driver element of the disk device includes the `cache="none"` and `io="native"` parameters.
+
+[source,terminal]
+----
+<disk type="block" device="disk">
+    <driver name="qemu" type="raw" cache="none" io="native" iothread="1"/>
+...
+</disk>
+----
+
+== Exclude the memory balloon device
+
+Unless you need a dynamic memory size, do not define a memory balloon device and ensure that libvirt does not create one for you. Include the `memballon` parameter as a child of the devices element in your domain configuration XML file.
+
+. Check the list of active profiles:
++
+[source,terminal]
+----
+<memballoon model="none"/>
+----
+
+== Tune the CPU migration algorithm of the host scheduler
+
+[IMPORTANT]
+====
+Do not change the scheduler settings unless you are an expert who understands the implications. Do not apply changes to production systems without testing them and confirming that they have the intended effect.
+====
+
+The `kernel.sched_migration_cost_ns` parameter specifies a time interval in nanoseconds. After the last execution of a task, the CPU cache is considered to have useful content until this interval expires. Increasing this interval results in fewer task migrations. The default value is 500000 ns.
+
+If the CPU idle time is higher than expected when there are runnable processes, try reducing this interval. If tasks bounce between CPUs or nodes too often, try increasing it.
+
+To dynamically set the interval to 60000 ns, enter the following command:
+
+[source,terminal]
+----
+# sysctl kernel.sched_migration_cost_ns=60000
+----
+
+To persistently change the value to 60000 ns, add the following entry to /etc/sysctl.conf:
+
+[source,terminal]
+----
+kernel.sched_migration_cost_ns=60000
+----
+
+== Disable the cpuset cgroup controller
+
+[NOTE]
+====
+This setting applies only to KVM hosts with cgroups version 1. To enable CPU hotplug on the host, disable the cgroup controller.
+====
+
+.Procedure
+
+. Open `/etc/libvirt/qemu.conf` with an editor of your choice.
+
+. Go to the `cgroup_controllers` line.
+
+. Duplicate the entire line and remove the leading number sign (#) from the copy. 
+
+. Remove the `cpuset` entry, as follows:
++
+[source,terminal]
+----
+cgroup_controllers = [ "cpu", "devices", "memory", "blkio", "cpuacct" ]
+----
+
+The new setting takes effect after you restart the libvirtd daemon:
+
+.Procedure
```

Comment:
To follow best practice, every section should only have a single procedure. As an alternative, perhaps this could be made into a step with several substeps?

```
. For the new setting to take effect, you must restart the libvirtd daemon.

.. Stop all virtual machines

.. Run the following command:

etc....
```